src/providers/aws.rs

```diff
@@ -827,6 +877,118 @@ impl RegionLauncher {
         Ok(self)
     }
 
+    #[instrument(level = "trace", skip(self))]
+    async fn make_on_demand_requests<M>(&mut self, machines: M) -> Result<(), Report>
```

Comment:
I wonder if it might be possible to share more of this code with the other implementation, but this seems fine.

Reply:
Maybe a few utility functions to e.g. do the instance grouping and make placement groups?

Reply:
Oh, yeah, probably. I'll do a pass and see if I can tidy it up a little.